.github/workflows/ci.yaml

```diff
@@ -142,3 +142,39 @@ jobs:
         with:
           name: docs
           path: docs/_build/html
+
+
+  api-backward-compatibilty:
+    name: Run Backward Compatibility Checks
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - id: base_branch
+        run: |
+            echo "::set-output name=branch::$(echo ${GITHUB_BASE_REF#refs/heads/})"
+      - name: Checkout Base Branch
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ steps.base_branch.outputs.branch }}
+          path: base/mlrun
+      - name: Checkout Head Branch and Stable with the target Branch
+        run : |
+          mkdir head && cd head
+          git init
+          git config --global user.name github-actions
+          git config --global user.email github-actions@mlrun
+          git config --global github.token ${{ secrets.GITHUB_TOKEN }}
+          git config --global pull.rebase false
+          git clone $GITHUB_SERVER_URL/$GITHUB_ACTOR/mlrun.git
+          cd mlrun
+          git remote add upstream $GITHUB_SERVER_URL/$GITHUB_REPOSITORY
+          git checkout $GITHUB_HEAD_REF
+          git pull upstream $GITHUB_BASE_REF --commit
+      - name : Setup python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+      - name: Install automation scripts dependencies and add mlrun to dev packages
+        run: cd head/mlrun && pip install -r requirements.txt -r dev-requirements.txt -r dockerfiles/mlrun-api/requirements.txt && pip install -e .
```

Comment:
The make file is missing having a `test-backward-compatibility-dockerized` target
when you'll add it, you won't need all this installation here